mediaengine.go

```diff
@@ -0,0 +1,206 @@
+	return codec.PayloadType
+}
+
+func (m *mediaEngine) getCodec(payloadType uint8) (*RTCRtpCodec, error) {
+	for _, codec := range m.codecs {
+		if codec.PayloadType == payloadType {
+			return codec, nil
+		}
+	}
+	return nil, errors.New("Codec not found")
+}
+
+func (m *mediaEngine) getCodecSDP(sdpCodec sdp.Codec) (*RTCRtpCodec, error) {
+	for _, codec := range m.codecs {
+		if codec.Name == sdpCodec.Name &&
+			codec.ClockRate == sdpCodec.ClockRate &&
+			(sdpCodec.EncodingParameters == "" ||
+				strconv.Itoa(int(codec.Channels)) == sdpCodec.EncodingParameters) &&
+			codec.SdpFmtpLine == sdpCodec.Fmtp { // TODO: Protocol specific matching?
+			return codec, nil
+		}
+	}
+	return nil, errors.New("Codec not found")
+}
+
+func (m *mediaEngine) getCodecsByKind(kind RTCRtpCodecType) []*RTCRtpCodec {
+	var codecs []*RTCRtpCodec
+	for _, codec := range rtcMediaEngine.codecs {
+		if codec.Type == kind {
+			codecs = append(codecs, codec)
+		}
+	}
+	return codecs
+}
+
+// NewRTCRtpOpusCodec is a helper to create an Opus codec
+func NewRTCRtpOpusCodec(payloadType uint8, clockrate uint32, channels uint16) *RTCRtpCodec {
+	c := NewRTCRtpCodec(RTCRtpCodecTypeAudio,
+		Opus,
+		clockrate,
+		channels,
+		"minptime=10;useinbandfec=1",
+		payloadType,
+		&codecs.OpusPayloader{})
+	return c
+}
+
+// NewRTCRtpVP8Codec is a helper to create an VP8 codec
+func NewRTCRtpVP8Codec(payloadType uint8, clockrate uint32) *RTCRtpCodec {
+	c := NewRTCRtpCodec(RTCRtpCodecTypeVideo,
+		VP8,
+		clockrate,
+		0,
+		"",
+		payloadType,
+		&codecs.VP8Payloader{})
+	return c
+}
+
+// NewRTCRtpVP9Codec is a helper to create an VP9 codec
+func NewRTCRtpVP9Codec(payloadType uint8, clockrate uint32) *RTCRtpCodec {
+	c := NewRTCRtpCodec(RTCRtpCodecTypeVideo,
+		VP9,
+		clockrate,
+		0,
+		"",
+		payloadType,
+		nil) // TODO
+	return c
+}
+
+// NewRTCRtpH264Codec is a helper to create an H264 codec
+func NewRTCRtpH264Codec(payloadType uint8, clockrate uint32) *RTCRtpCodec {
+	c := NewRTCRtpCodec(RTCRtpCodecTypeVideo,
+		H264,
+		clockrate,
+		0,
+		"level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f",
+		payloadType,
+		&codecs.H264Payloader{})
+	return c
+}
+
+// RTCRtpCodecType determines the type of a codec
+type RTCRtpCodecType int
+
+const (
+
+	// RTCRtpCodecTypeAudio indicates this is an audio codec
+	RTCRtpCodecTypeAudio RTCRtpCodecType = iota + 1
+
+	// RTCRtpCodecTypeVideo indicates this is a video codec
+	RTCRtpCodecTypeVideo
+)
+
+func (t RTCRtpCodecType) String() string {
+	switch t {
+	case RTCRtpCodecTypeAudio:
+		return "audio"
+	case RTCRtpCodecTypeVideo:
+		return "video"
+	default:
+		return "Unknown"
+	}
+}
+
+// RTCRtpCodec represents a codec supported by the PeerConnection
+type RTCRtpCodec struct {
+	RTCRtpCodecCapability
+	Type        RTCRtpCodecType
+	Name        string
+	PayloadType uint8
+	Payloader   rtp.Payloader
+}
+
+// NewRTCRtpCodec is used to define a new codec
+func NewRTCRtpCodec(
+	typ RTCRtpCodecType,
```

Comment:
`typ -> type` just easier to read in GoDoc

Reply:
'type' is a keyword thought. Maybe codecType?